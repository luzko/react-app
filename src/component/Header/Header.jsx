import React, {useState} from 'react';
import style from './Header.module.css';
import Search from "../Search";
import AddMovie from "../AddMovie";
import logo from '../../../public/logo.svg';
import {navigateToSearch} from "../../helper/routeHelper";
import {useNavigate, useParams, useSearchParams} from "react-router-dom";

const Header = (props) => {
  let navigate = useNavigate();
  let [searchParams] = useSearchParams();
  let {searchQuery} = useParams();
  let [search, setSearch] = useState(searchQuery ? searchQuery : "")

  let handleSearch = () => {
    navigateToSearch(navigate, search, searchParams)
  };

  const keyEnter = (e) => {
    if (e.which === 13) {
      handleSearch();
    }
  };

  return (
      <header className={style.header}>
        <div className={style.logoAndButton}>
          <img src={logo} alt="logo" className={style.logo}/>
          <AddMovie createMovie={props.createMovie}/>
        </div>
        <Search
            searchValue={search}
            setSearchValue={setSearch}
            handleSearch={handleSearch}
            keyEnter={keyEnter}
        />
      </header>
  );
};

export default Header;
