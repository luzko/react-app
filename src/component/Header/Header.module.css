.header {
  background: linear-gradient(rgba(0, 0, 0, 1), rgba(0, 0, 0, 0.6)),
  url('https://miro.medium.com/max/1400/1*5lyavS59mazOFnb55Z6znQ.png');
  padding: 10px;
}

.logoAndButton {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 20px;
}

.logo {
  width: 110px;
}
