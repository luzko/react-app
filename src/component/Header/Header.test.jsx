import React from "react";
import {fireEvent, render, screen} from "@testing-library/react";
import {navigateToSearch} from "../../helper/routeHelper";
import Header from "../Header";

jest.mock('../../helper/routeHelper')

jest.mock("react-router-dom", () => {
  return {
    useHistory: jest.fn(),
    useSearchParams: jest.fn(() => [{}]),
    useParams: jest.fn(() => ({
      searchQuery: ""
    }))
  }
})

it("header search test", () => {
  const {asFragment} = render(<Header/>)
  expect(asFragment()).toMatchSnapshot()

  fireEvent.change(screen.getByPlaceholderText('What do you want to watch?'), {
    target: {value: 'name'}
  })

  expect(screen.getByDisplayValue('name')).toBeInTheDocument
  fireEvent.click(screen.getByTitle('search'))
  expect(navigateToSearch).toHaveBeenCalled()
})
