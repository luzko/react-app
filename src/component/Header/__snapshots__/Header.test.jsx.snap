// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header search test 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="logoAndButton"
    >
      <img
        alt="logo"
        class="logo"
        src="logo.svg"
      />
      <button
        class="addButton"
      >
        + ADD MOVIE
      </button>
      <div
        class="modal "
      >
        <div
          class="content"
        >
          <button
            class="close"
          >
            ✕
          </button>
          <p
            class="title"
          >
            ADD MOVIE
          </p>
          <form>
            <div
              class="field"
            >
              <label>
                TITLE
              </label>
              <input
                class="input "
                id="title"
                name="title"
                type="text"
                value=""
              />
            </div>
            <div
              class="field"
            >
              <label>
                TAGLINE
              </label>
              <input
                class="input "
                id="tagline"
                name="tagline"
                type="text"
                value=""
              />
            </div>
            <div
              class="field"
            >
              <label>
                RELEASE DATE
              </label>
              <input
                class="input "
                id="release"
                name="release"
                type="date"
                value=""
              />
            </div>
            <div
              class="field"
            >
              <label>
                POSTER
              </label>
              <input
                class="input "
                id="poster"
                name="poster"
                type="text"
                value=""
              />
            </div>
            <div
              class="field"
            >
              <label>
                RATING
              </label>
              <input
                class="input "
                id="rating"
                name="rating"
                type="text"
                value=""
              />
            </div>
            <div
              class="field"
            >
              <label>
                GENRES
              </label>
              <input
                class="input "
                disabled=""
                id="genre"
                name="genre"
                type="text"
                value=""
              />
              <div
                class="arrow"
              >
                ▼
              </div>
              <div
                class="list hidden"
              >
                <li
                  class="li"
                >
                  <input
                    name="genre"
                    type="checkbox"
                    value="Fantasy"
                  />
                  <label>
                    Fantasy
                  </label>
                </li>
                <li
                  class="li"
                >
                  <input
                    name="genre"
                    type="checkbox"
                    value="Drama"
                  />
                  <label>
                    Drama
                  </label>
                </li>
                <li
                  class="li"
                >
                  <input
                    name="genre"
                    type="checkbox"
                    value="Action"
                  />
                  <label>
                    Action
                  </label>
                </li>
                <li
                  class="li"
                >
                  <input
                    name="genre"
                    type="checkbox"
                    value="Comedy"
                  />
                  <label>
                    Comedy
                  </label>
                </li>
                <li
                  class="li"
                >
                  <input
                    name="genre"
                    type="checkbox"
                    value="Crime"
                  />
                  <label>
                    Crime
                  </label>
                </li>
                <li
                  class="li"
                >
                  <input
                    name="genre"
                    type="checkbox"
                    value="Adventure"
                  />
                  <label>
                    Adventure
                  </label>
                </li>
                <li
                  class="li"
                >
                  <input
                    name="genre"
                    type="checkbox"
                    value="Thriller"
                  />
                  <label>
                    Thriller
                  </label>
                </li>
                <li
                  class="li"
                >
                  <input
                    name="genre"
                    type="checkbox"
                    value="Family"
                  />
                  <label>
                    Family
                  </label>
                </li>
                <li
                  class="li"
                >
                  <input
                    name="genre"
                    type="checkbox"
                    value="Documentary"
                  />
                  <label>
                    Documentary
                  </label>
                </li>
                <li
                  class="li"
                >
                  <input
                    name="genre"
                    type="checkbox"
                    value="Horror"
                  />
                  <label>
                    Horror
                  </label>
                </li>
              </div>
            </div>
            <div
              class="field"
            >
              <label>
                RUNTIME
              </label>
              <input
                class="input "
                id="runtime"
                name="runtime"
                type="text"
                value=""
              />
            </div>
            <div
              class="field"
            >
              <label>
                OVERVIEW
              </label>
              <textarea
                class="input "
                id="overview"
                name="overview"
                type="text"
              />
            </div>
            <div
              class="buttons"
            >
              <button
                class="button"
              >
                RESET
              </button>
              <button
                class="button"
                type="submit"
              >
                SUBMIT
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
    <div
      class="headerText"
    >
      FIND YOUR MOVIE
    </div>
    <div
      class="search"
    >
      <input
        class="searchInput"
        id="search"
        placeholder="What do you want to watch?"
        type="text"
        value=""
      />
      <button
        class="searchButton"
        title="search"
      >
        SEARCH
      </button>
    </div>
  </header>
</DocumentFragment>
`;
